### PR TITLE
Remove eval from Lightbox

### PR DESCRIPTION
**Pull request by crhallberg**
In the Lightbox, eval has been an available option for custom callbacks. This was so developers can treat the `data-lightbox-onsubmit` and other callbacks as the regular `onsubmit`, with a choice of function names or inline Javascript.

This PR removes the one instance of eval usage in the core code and removes the eval option. Custom functions can still be created and called, but inline Javascript is not longer valid.

### CODE REVIEW

File: themes/bootstrap3/js/lightbox.js [hunk 1]
```diff
@@ -184,8 +184,16 @@ VuFind.register('lightbox', function Lightbox() {
   function _evalCallback(callback, event, data) {
     if ('function' === typeof window[callback]) {
       return window[callback](event, data);
+    } else if (callback.substr(0, 7) === 'VuFind.') {
+      var parts = callback.substr(7).split('.');
+      var obj = VuFind;
+      for (var i = 0; i < parts.length; i++) {
+        obj = obj[parts[i]];
+      }
+      return obj(event, data);
```
Comment (demiankatz) [inline themes/bootstrap3/js/lightbox.js]: Should you check the type of obj before doing this?

File: themes/bootstrap3/js/lightbox.js [hunk 2]
```diff
@@ -198,9 +206,7 @@ VuFind.register('lightbox', function Lightbox() {
    * data-lightbox-title = Lightbox title (overrides any title the page provides)
    */
   _constrainLink = function constrainLink(event) {
-    if (typeof $(this).data('lightboxIgnore') != 'undefined'
-      || typeof this.attributes.href === 'undefined' || this.attributes.href.value.charAt(0) === '#'
-    ) {
+    if (typeof $(this).data('lightboxIgnore') != 'undefined' || this.attributes.href.value.charAt(0) === '#') {
```
Comment (demiankatz) [inline themes/bootstrap3/js/lightbox.js]: This appears to be reverting Ere's fix from yesterday.

**Comment by crhallberg** [comment]
Hello, @EreMaijala! This is ready to go, but we wanted to get your eyes on it, you seem to have a knack for catching JS problems. We did this to remove eval from the JS, making the Lightbox just a little more secure and making our integrated testing just a little happier.

Hope the new Request Review feature worked well.

File: themes/bootstrap3/js/lightbox.js [hunk 3]
```diff
@@ -184,9 +184,23 @@ VuFind.register('lightbox', function Lightbox() {
   function _evalCallback(callback, event, data) {
     if ('function' === typeof window[callback]) {
       return window[callback](event, data);
-    } else {
-      return eval('(function(event, data) {' + callback + '}())'); // inline code
     }
+    if (callback.substr(0, 7) === 'VuFind.') {
+      var parts = callback.substr(7).split('.');
```
Comment (crhallberg) [inline themes/bootstrap3/js/lightbox.js]: This general approach is much better. I'll work it in. Thank you!

**Comment by demiankatz** [comment]
All tests passing -- merging now. Thanks, everyone!

**Pull request merged**